From 74f82b86b8d903e37de60d31b81c698702219c93 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Tue, 16 Dec 2014 11:26:22 +0800
Subject: [PATCH 192/196] atomisp2: Enable YUVPP pipe

This patch enables YUVPP pipe

Change-Id: I894345afaa79fb3f73da6ced2bea633c88dd3f26
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-13815
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/32926
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_internal.h   | 20 +++++---------------
 1 file changed, 5 insertions(+), 15 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
index 42cb36d..21edf15 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_internal.h
@@ -150,25 +150,12 @@
  */
 #define ATOMISP_MAX_ISR_LATENCY	1000
 
-/*
- * Add new YUVPP pipe for SOC sensor.
- * a.ATOMISP_CSS_SUPPORT_YUVPP = 1
- *    the css support YUVPP for SOC sensor.
- * b.ATOMISP_CSS_SUPPORT_YUVPP = 0
- *    the css no support YUVPP for SOC sensor
- * c.Now, the FW has some issue about YUVPP pipe. so I disable YUVPP pipe for
- *   SOC sensor
- */
-#define ATOMISP_CSS_SUPPORT_YUVPP     0
+/* Add new YUVPP pipe for SOC sensor. */
+#define ATOMISP_CSS_SUPPORT_YUVPP     1
 
 #define ATOMISP_CSS_OUTPUT_SECOND_INDEX     1
 #define ATOMISP_CSS_OUTPUT_DEFAULT_INDEX    0
 
-#define ATOMISP_USE_YUVPP(asd)  \
-	(asd->isp->inputs[asd->input_curr].type == SOC_CAMERA \
-	&& asd->isp->inputs[asd->input_curr].camera_caps-> \
-	   sensor[asd->sensor_curr].stream_num == 1   \
-	&& ATOMISP_CSS_SUPPORT_YUVPP)
 /*
  * ATOMISP_SOC_CAMERA
  * This is to differentiate between ext-isp and soc camera in
@@ -179,6 +166,9 @@
 	&& asd->isp->inputs[asd->input_curr].camera_caps-> \
 	   sensor[asd->sensor_curr].stream_num == 1)
 
+#define ATOMISP_USE_YUVPP(asd)  \
+	(ATOMISP_SOC_CAMERA(asd) && ATOMISP_CSS_SUPPORT_YUVPP)
+
 #define ATOMISP_DEPTH_SENSOR_STREAMON_COUNT 2
 
 #define DIV_NEAREST_STEP(n, d, step) \
-- 
1.9.1

